### PR TITLE
add show instance for Encoding

### PR DESCRIPTION
**Pull request by obadz**
Written during http://www.meetup.com/London-Haskell-Hacking/events/231965828/ :-)

cc @dcoutts 

(Edited meetup link which was wrong. ht @nmattia :-))

### CODE REVIEW

**Comment by thoughtpolice** [comment]
hs-boot files make me feel very not-:100:, tbh. I'm discussing with @dcoutts, and I'll relay more here later, but, in short:

  1) We may want to move stuff around a bit anyway, despite this, meaning we might be able to avoid some of the recursive dependencies.
  2) While we definitely agree `Show` is a necessary addition (it's so convenient), it's not clear *what* representation should be used, based on its purpose. Should we use `deriving Show`, `show . toFlatTerm` or `show . ... . decodeTerm`?

So, in short, we do really appreciate the PR! Now we have to think about a few things first, but I'm leaving this open because I will come back to it.

**Comment by obadz** [comment]
@thoughtpolice, I completely understand. @dcoutts & I went over all these points yesterday during the meetup and he warned me that this might not get merged but I thought it would be fun for me to learn about `hs-boot`. I definitely won't take it personally if this gets dropped :-)

**Comment by dcoutts** [comment]
We've concluded this is actually the least-worst solution :-)

**Comment by obadz** [comment]
:+1: :-)
